src/number.ts

```diff
@@ -14,6 +14,7 @@ const approx =
         `${mantissa}e${(exponent ? Number(exponent) : 0) - digits}`,
       Number,
       (n) => Math.sign(a) * Math.abs(n), // Helps when rounding -0.
+      (n) => (n === 0 && discardZeroSign ? Math.abs(n) : n), // If true, -0 -> 0
```

Comment:
I think line 16 is used to preserve signed zero. You should be able to get the same result discarding your line and making mine conditional.

Reply:
I think it's being quite difficult to read, for that reason is treated as a case apart, but I also like one-line functions, so ok!

Reply:
Your choice then, if you prefer reverting it's fine!